tests/test_surfaces.py

```diff
@@ -136,6 +137,140 @@ def test_curvature(self):
+                    data_offset["x"][:, 0],
+                    np.zeros_like(grid_compute.nodes[:, 1]),
+                )
+            ).T,
+        }
+        for key in ["e_theta", "e_zeta"]:
+            np.testing.assert_allclose(
+                correct_data_offset[key],
+                data_offset[key],
+                atol=1e-4,
+                err_msg=f"Failed test at comparison of {key}",
+            )
+
+    @pytest.mark.slow
+    @pytest.mark.unit
+    def test_constant_offset_surface_rot_ellipse(self):
+        """Test constant offset algorithm for a rotating ellipse."""
+        eq = desc.examples.get("HELIOTRON")
+        s = eq.surface
+        grid = LinearGrid(M=4, N=4, NFP=eq.NFP)
+        offset = 0.1
+        (s_offset, data, _) = FourierRZToroidalSurface.constant_offset_surface(
+            s, offset, grid, M=2, N=2, full_output=True
+        )
+        r_offset_surf = data["x_offset_surface"]
+        r_surf = data["x"]
+
+        dists = np.linalg.norm(rpz2xyz(r_surf) - rpz2xyz(r_offset_surf), axis=1)
+
+        np.testing.assert_allclose(dists, offset, atol=1e-16)
+
+        R00_offset_ind = s_offset.R_basis.get_idx(M=0, N=0)
+        R00_offset = s_offset.R_lmn[R00_offset_ind]
+
+        R10_offset_ind = s_offset.R_basis.get_idx(M=1, N=0)
+        R10_offset = s_offset.R_lmn[R10_offset_ind]
+        R11_offset_ind = s_offset.R_basis.get_idx(M=1, N=1)
+        R11_offset = s_offset.R_lmn[R11_offset_ind]
+        Rneg1neg1_offset_ind = s_offset.R_basis.get_idx(M=-1, N=-1)
+        Rneg1neg1_offset = s_offset.R_lmn[Rneg1neg1_offset_ind]
+
+        Zneg10_offset_ind = s_offset.Z_basis.get_idx(M=-1, N=0)
+        Zneg10_offset = s_offset.Z_lmn[Zneg10_offset_ind]
+        Zneg11_offset_ind = s_offset.Z_basis.get_idx(M=-1, N=1)
+        Zneg11_offset = s_offset.Z_lmn[Zneg11_offset_ind]
+        Z1neg1_offset_ind = s_offset.Z_basis.get_idx(M=1, N=-1)
+        Z1neg1_offset = s_offset.Z_lmn[Z1neg1_offset_ind]
+
+        # a is the cos or sin (theta) term for the ellipse
+        # b is the cos(theta)*sin(phi) mixed terms for the ellipse
+        a = 1
+        b = 0.3
+
+        np.testing.assert_allclose(
+            R00_offset,
+            eq.surface.R_lmn[eq.surface.R_basis.get_idx(M=0, N=0)],
+            atol=1e-3,
+        )
+        np.testing.assert_allclose(R10_offset, -a - offset, atol=1e-2)
+        np.testing.assert_allclose(R11_offset, -b, atol=3e-2)
+        np.testing.assert_allclose(Rneg1neg1_offset, b, atol=3e-2)
+
+        np.testing.assert_allclose(Zneg10_offset, a + offset, atol=1e-2)
+        np.testing.assert_allclose(Zneg11_offset, -b, atol=3e-2)
+        np.testing.assert_allclose(Z1neg1_offset, -b, atol=3e-2)
+
+        np.testing.assert_allclose(
+            R00_offset,
+            eq.surface.R_lmn[eq.surface.R_basis.get_idx(M=0, N=0)],
+            atol=1e-3,
+        )
+        # cannot do the same sort of distance test as the axisymmetric test
```

Comment:
Could maybe compute aspect ratio and volume before and after?

Reply:
added a XS surface area test